### PR TITLE
mrc-3272 compile ODE models to mode 

### PR DESCRIPTION
**Pull request by hillalex**
Adds `mode` as a compilation target

### CODE REVIEW

File: tests/testthat/test-odin-dust.R [hunk 1]
```diff
@@ -494,12 +483,55 @@ test_that("Detect sum corner case", {
 })
 
 
-test_that("disallow output()", {
```
Comment (hillalex) [inline tests/testthat/test-odin-dust.R:497]: this just looked like a dupe

File: tests/testthat/test-odin-dust.R [hunk 2]
```diff
@@ -494,12 +483,62 @@ test_that("Detect sum corner case", {
...
+  actual <- vapply(times, function(t) mod$run(t),
+                   matrix(0.0, 3, n_particles))
+  expect_equal(actual[1:2, 1, ], analytic, tolerance = 1e-7)
+})
+
+
+test_that("correctly compiles compartmental model", {
+  gen <- odin_dust_("examples/age.R")
+  mod <- gen$new(list(IO = 1), 0, 1)
+  expect_silent(mod$run(1))
```
Comment (richfitz) [inline tests/testthat/test-odin-dust.R:543]:
```suggestion
  cmp <- odin::odin("examples/age.R", target = "c")$new(I0 = 1, use_dde = TRUE)
  y_mode <- mod$run(10)
  y_odin <- cmp$run(c(0, 10))[2, -1]
  expect_equal(drop(y_mode), unname(y_odin))
```

File: R/generate_dust.R [hunk 3]
```diff
@@ -383,15 +445,19 @@ dust_unpack_variable <- function(name, dat, state, rewrite) {
...
     sprintf("%s[%s]", state, rewrite(x$offset))
   } else {
     ## Using a wrapper here would be more C++'ish but is it needed?
     offset <- rewrite(x$offset)
-    len <- rewrite(d$dimnames$length)
-    sprintf("%s + %s", state, offset)
+    if (discrete) {
+      sprintf("%s + %s", state, offset)
+    }
+    else {
```
Comment (richfitz) [inline R/generate_dust.R:458]: sorry, tiny thing here - I can also update it in the following PR

**Review by richfitz — CHANGES_REQUESTED** [review]
Looks good - just added a suggestion for a slightly better test of that age structured model

File: R/generate_dust.R (hunk 3 above)
Comment (hillalex) [inline R/generate_dust.R:458]:
```suggestion
    } else {
```

**Review by richfitz — APPROVED** [review]
